src/application.rs

```diff
@@ -106,6 +119,41 @@ impl Application {
 
         loop {
             match self.event_queue.receive() {
+                Event::ReadFile(chunk) => {
+                    let try_send = || -> Result<()> {
+                        let Chunk {
+                            id,
+                            file_name,
+                            data,
+                            bytes_read,
+                            file_size,
+                        } = chunk?;
+
+                        self.network
+                            .send_all(
+                                state.all_user_endpoints(),
+                                NetMessage::UserData(file_name, Some((data, bytes_read)), None),
+                            )
+                            .map_err(stringify_sendall_errors)?;
+
+                        if bytes_read == 0 {
+                            state.progress_stop(id);
+                        } else {
+                            state.progress_pulse(id, file_size, bytes_read);
+                        }
+                        Ok(())
+                    };
+
+                    if let Err(e) = try_send() {
+                        // we dont have the file_name here
+                        // we'll just stop the last progress
+                        state.progress_stop_last();
+                        let msg = format!("Error sending file. error: {}", e);
+                        state.add_message(termchat_message(msg, TermchatMessageType::Error));
+                    }
+
+                    ui::draw(&mut self.terminal, &state)?;
```

Comment:
I think that you do not need this line, because it will be called at the end of the loop in line 338

Reply:
yes I missed that.